File: src-tauri/src/core/entities/ministry_event.rs
use chrono::prelude::*;
use serde::{Deserialize, Serialize};
use specta::Type;

#[derive(Debug, Serialize, Deserialize, Type)]
#[serde(rename_all = "camelCase")]
pub struct MinistryEvent {
    #[specta(type = i32)]
    pub id: i64,
    pub assignee_name: String,
    #[specta(type = Option<i32>)]
    pub assignee_id: Option<i64>,
    // pub scheduled_time: DateTime<Utc>,
    pub date: NaiveDate,
    pub time: Option<NaiveTime>,
    pub place: String,
    pub extra_info: String,
}

#[derive(Debug, Serialize, Deserialize, Type)]
#[serde(rename_all = "camelCase")]
pub struct NewMinistryEvent {
    pub assignee_name: String,
    #[specta(type = Option<i32>)]
    pub assignee_id: Option<i64>,
    // pub scheduled_time: DateTime<Utc>,
    pub date: NaiveDate,
    pub time: Option<NaiveTime>,
    pub place: String,
    pub extra_info: String,
}

// impl From<

File: src/routes/planning/MinistryEvent.ts
import type { MinistryEvent as MinistryEventDto } from 'bindings';

export class MinistryEvent {
	id: number;
	scheduledTime: Date;
	place: string;
	extraInfo: string;
	assigneeName: string;
	assigneeId?: number;

	constructor(fromData: MinistryEventDto) {
		this.id = fromData.id;
		this.scheduledTime = new Date(Date.parse(fromData.scheduledTime));
		this.place = fromData.place;
		this.extraInfo = fromData.extraInfo;
		(this.assigneeName = fromData.assigneeName),
			(this.assigneeId = fromData.assigneeId ?? undefined);
	}

	asDto(): MinistryEventDto {
		return {
			id: this.id,
			assigneeId: this.assigneeId ?? null,
			place: this.place,
			assigneeName: this.assigneeName,
			extraInfo: this.extraInfo,
			scheduledTime: this.scheduledTime.toISOString()
		};
	}
}
